__tests__/workers/postUpdated.ts

```diff
@@ -0,0 +1,107 @@
+import { expectSuccessfulBackground, saveFixtures } from '../helpers';
+import worker from '../../src/workers/postUpdated';
+import {
+  ArticlePost,
+  Post,
+  PostOrigin,
+  PostType,
+  SharePost,
+  Source,
+} from '../../src/entity';
+import { sourcesFixture } from '../fixture/source';
+import { DataSource } from 'typeorm';
+import createOrGetConnection from '../../src/db';
+
+let con: DataSource;
+
+beforeAll(async () => {
+  con = await createOrGetConnection();
+});
+
+beforeEach(async () => {
+  jest.resetAllMocks();
+  await saveFixtures(con, Source, sourcesFixture);
+  await saveFixtures(con, ArticlePost, [
+    {
+      id: 'p1',
+      shortId: 'p1',
+      url: 'http://p1.com',
+      score: 0,
+      metadataChangedAt: new Date('01-05-2020 12:00:00'),
+      sourceId: 'a',
+      visible: false,
+      createdAt: new Date('01-05-2020 12:00:00'),
+      origin: PostOrigin.Squad,
+    },
+  ]);
+});
+
+const createSharedPost = async (id = 'sp1') => {
+  const post = await con.getRepository(Post).findOneBy({ id: 'p1' });
+  await con.getRepository(SharePost).save({
+    ...post,
+    id,
+    shortId: `short-${id}`,
+    sharedPostId: 'p1',
+    type: PostType.Share,
+    visible: false,
+  });
+};
+
+it('should not update if the database updated date is newer', async () => {
+  await expectSuccessfulBackground(worker, {
+    updated_at: new Date('01-05-1990 12:00:00'),
+  });
+  const post = await con.getRepository(ArticlePost).findOneBy({ id: 'p1' });
+  expect(post.metadataChangedAt).toEqual(new Date('2020-01-05T12:00:00.000Z'));
+});
+
+it('should not update if the post is not a squad origin', async () => {
+  await con
+    .getRepository(ArticlePost)
+    .update({ id: 'p1' }, { origin: PostOrigin.CommunityPicks });
+  await expectSuccessfulBackground(worker, {
+    updated_at: new Date('01-05-1990 12:00:00'),
```

Comment:
An id is missing here that indicate what post to look at. And updated_at is too old compared to the metadata changed at